### ISSUE TITLE
Clarify what yarn *is*

### ISSUE THREAD

**Issue by fingolfin** [yarnpkg/yarn#1046]
Right now, this is described as "Fast, reliable, and secure dependency management". But it seems this is really for Node.js and/or Javascript... I really am not sure, tried for some time to figure it out from your github repo and your website, but frustratingly, it necer really seems to say it...

I understand it's hardly a priority for you (as your users of course know the answer), but anybody else who stumbles over your project will appreciate it :-)

**Comment by cpojer** [comment]
Yarn is language agnostic and we are planning to use it for other languages in the future. You are right, currently the main use case is to manage dependencies for JavaScript but there is very little about it that ties it to the JavaScript ecosystem. "Fast, reliable, and secure dependency management" is really the ideal description of what Yarn does for you. We have on occasion messaged it as "for JavaScript" but our longterm vision is that JavaScript will be just one of the many languages that Yarn supports.

Please send us a PR that will help make the website more clear but keeps the vision we have for the project and we can discuss it there.

Also see https://github.com/yarnpkg/website/issues/133#issuecomment-253179757

**Comment by fingolfin** [comment]
Well, if yarn is meant to be generic like that, then *this* information would be helpful.

From the little I could gather from reading your website, "getting started", and some other parts of the doc this was not apparent; I got the distinct impression that you make a lot of (silent) assumptions, which finally lead me to the guess that this is about Javascript / Node (some googline seemed to confirm that).

There are other things that are unclear. Like, "Getting Started" says that you can specify "dependencies" -- but what values are allowed there? 

In closing, asking me (somebody who knows nothing about yarn) to send a PR to improve your website "but keep the vision" (what *is* the vision? how would I even know what it is? I am asking precisely because that is unclear) really feels like you are telling me "shut up and f**k of" :-(.

**Comment by fingolfin** [comment]
BTW, the information on https://code.facebook.com/posts/1840075619545360?_fb_noscript=1 was several order of magnitudes more enlightning and helpful than anything I found on your website or in your documentation.

**Comment by cpojer** [comment]
That was certainly not my intention. I stated above that we'd like Yarn to be a packager manager for many languages. The Yarn project is supposed to be completely open and I asked you to help clarify the wording on the website as an invitation, not as a way to ask you to shut up. Your feedback is very valuable and if you are confused, chances are that other people are, too. I was thinking that your confusion and journey to understand what Yarn is would help you document it better for others.

We scrambled a bit to get the website and documentation together prior to the release and I agree that we can improve most of it. We are happy to accept any PRs on the website that will help us clarify our goals. Right now, the team at FB is focused on getting everything figured out post launch. We didn't expect this huge amount of contributions and it's hard to keep up so we haven't been able to spend more time on the website yet.
